### PR TITLE
Fetch data from Firebase Database

### LINKED ISSUE #29: Fetch data from Firebase Db
We like to connect to the Firebase and fetch data for the app from Firebase DB.

Check the following URLs.
1. https://project-7593821828655173684.firebaseio.com/Categories.json
2. https://project-7593821828655173684.firebaseio.com/Sites.json
3. https://project-7593821828655173684.firebaseio.com/Sites/Newspapers.json

Current database structure is like this 

image

This file contains all the relevant details to the Firebase. The database is public readable.
https://github.com/CodeLanka/ez-net-app/blob/master/android/app/google-services.json

### PR DESCRIPTION
**Pull request by TheLukaszNs**
What this PR can help with?
===
Well, this PR fixes #29 and adds new **Api class** which can be developed along with adding new Firebase functionality.

But wait!
===
I think we need to change/rebuild Firebase database structure (and then some of `Api.js`), because it doesn't contain important information (like `thumbnail`) or is structured incorrectly (for example `Sites/Newspapers` returns:
```json
{
 "දිවයින":{"url":"www.divaina.com"},
 "ලංකාදීප":{"url":"www.lankadeepa.lk"}
}
```
instead of (like it's in dummy_api.js):
```json
[
 {"id": 1, "title":"දිවයින", "thumbnail": "NEWSPAPERS_DIVAINA", "url":"www.divaina.com},
 .
 .
]
```
(I know it's hard to make such a data structure in Firebase 😞) so I had to correct it manually in the code which makes it a little bit less reader firendly). There's one more bug: look at line 44 in `Api.js`

***
I'm waiting for review, for now: ***Happy Xmas!*** 🎄

> Yeah, it **probably** needs a bigger review and next patch to fix this Database structure

### CODE REVIEW

**Comment by TheLukaszNs** [comment]
And one more important note: I had tested it only on Android

File: src/components/App.js [hunk 1]
```diff
@@ -36,15 +36,18 @@ class Categories extends Component {
...
 
-    const nodes = this.props.categoryData.map(category => (
-      <BoxItem
-        key={category.id}
-        onPress={() => navigate('Detail', { title: category.title, items: category.items })}
-        {...category}
-      />
-    ));
+    let nodes = null;
+    if (this.props.categoryData.length !== 0) {
```
Comment (agentmilindu) [inline src/components/App.js]: is this check necessary?

**Comment by agentmilindu** [comment]
Wonderful job!

**Comment by agentmilindu** [comment]
@nushydude Hope you too can go through this.

**Comment by TheLukaszNs** [comment]
I think that database is now structured better, but before I push next commit to my PR: wouldn't it be nice if users could see categories icons, too? We should add thumbnails fields to categories, because these questions marks isn't looking so good. Take a look at this [gist](https://gist.github.com/TheLukaszNs/2b4f2590709698e235427a40c890e560).

___
Also, in my opinion, it'd be much better for code when the URLs had contain `https://` or `http://` prefixes.

File: src/components/App.js (hunk 1 above)
Comment (TheLukaszNs) [reply src/components/App.js]: Yeah, that isn't necessary, I'll delete it in next commit 😄!

**Comment by agentmilindu** [comment]
@TheLukaszNs yes, indeed, let's add category thumbnails too. I will add the HTTP protocols to the URLs also.

**Comment by TheLukaszNs** [comment]
@agentmilindu I had updated `Api.js` with new Firebase fields, added temporary `CATEGORY_COOKING` image (**CC0** license) and deleted this unnecessary check in `App.js`.

**Comment by agentmilindu** [comment]
Awesome! 
BTW, the PR says me there is a conflict in `dummy_api.js`, thus I can't merge this. Can you resolve the conflict, please?

**Pull request merged**